model-optimizer/extensions/ops/gather_test.py

```diff
@@ -3,59 +3,168 @@
-            'gather_input3_data': {'shape': None, 'value': 0, 'kind': 'data'},
+        graph = build_graph(nodes, edges)
+        graph.stage = 'middle'
+        partial_infer(graph)
 
-            'gather_node': {'op': 'Gather', 'kind': 'op'},
-            'gather_output': {'shape': None, 'value': None, 'kind': 'data'}
+        node = Node(graph, 'gather')
+        res = node.out_port(0).data.get_value()
+        npt.assert_array_equal(res, ref_value)
 
+    @staticmethod
+    def build_and_test_shape_inference(data_shape, indices_shape, axis, batch_dims, ref_shape):
+        nodes = {
+            **shaped_input('data', int64_array(data_shape)),
+            **shaped_input('indices', int64_array(indices_shape)),
+            **valued_const_with_data('axis', int64_array(axis)),
+            **regular_op_with_empty_data('gather', {'op': 'Gather', 'batch_dims': batch_dims, 'infer': Gather.infer}),
+            **result('res'),
         }
-        return build_graph(nodes_attributes,
-                           [
-                               ('gather_input', 'gather_input_data'),
-                               ('gather_input2', 'gather_input2_data'),
-                               ('gather_input3', 'gather_input3_data'),
-
-                               ('gather_input_data', 'gather_node'),
-                               ('gather_input2_data', 'gather_node'),
-                               ('gather_input3_data', 'gather_node'),
-
-                               ('gather_node', 'gather_output'),
-                           ],
-                           {
-                               'gather_input_data': {'shape': int64_array([10, 15]), 'value': np.ones((3, 15))},
-                               'gather_input2_data': {'shape': int64_array([2]), 'value': np.array([0, 2])},
-                           })
-
-    def test_gather_infer(self):
-        graph = self._create_graph()
-
-        gather_node = Node(graph, 'gather_node')
-        Gather.infer(gather_node)
-
-        exp_shape = int64_array([2, 15])
-        res_shape = graph.node['gather_output']['shape']
-        res_value = graph.node['gather_output']['value']
-
-        self.assertTrue(np.array_equal(exp_shape, res_shape),
-                        'shapes do not match expected: {} and given: {}'.format(exp_shape, res_shape))
-
-        self.assertTrue(np.array_equal(res_value, np.ones(exp_shape)),
-                        'shapes do not match expected: {} and given: {}'.format(exp_shape, res_shape))
+
+        edges = [
+            *connect('data', '0:gather'),
+            *connect('indices', '1:gather'),
+            *connect('axis', '2:gather'),
+            *connect('gather', 'res')
+        ]
+
+        graph = build_graph(nodes, edges)
+        graph.stage = 'middle'
+        partial_infer(graph)
+
+        node = Node(graph, 'gather')
+        res = node.out_port(0).data.get_shape()
+        npt.assert_array_equal(res, ref_shape)
+
+    def test_shape_axis_1(self):
```

Comment:
May be use unittest generator?

Reply:
I think it's better to not use them here.
It's not possibly to specify keyword arguments while using test generators. This makes more difficult to write them and understand relationship between formal and actual parameters of a function.

https://github.com/openvinotoolkit/openvino/blob/master/model-optimizer/unit_tests/mo/ops/broadcast_test.py#L22-L46
https://github.com/openvinotoolkit/openvino/blob/master/model-optimizer/unit_tests/mo/ops/slice_test.py#L20-L43

If i write by hand then I can leave comment (usually with keyword names and actual values) below test function 
https://github.com/openvinotoolkit/openvino/blob/master/model-optimizer/unit_tests/mo/ops/strided_slice_test.py#L1170-L1178
and PyCharm will show this description, this also enhances localizing if something went wrong 
![image](https://user-images.githubusercontent.com/5703039/116059416-7b0b4a80-a689-11eb-8498-2566dd98243d.png)


The only economy we get from `test-generator` library is lesser lines of code but disadvantages are:
- more difficult to write
- difficult to understand
- the `test-generator` library is not so widely used and maintained https://pypi.org/project/test-generator/, so it's better not to be bound to this library